### PR TITLE
Homework5: Olga Chubryk

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/hometask5/runner.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+def runner(*functions):
+    if not functions:
+        functions = [el for el in dir(the_functions) if el[:2]
+                     not in ['re', '__']]
+    for function in functions:
+        called_function = getattr(the_functions, function)
+        print('Вызов функции:', function)
+        print('Результат:', called_function())
```
Comment (belovmd) [inline src/hometask5/runner.py]: we should check if it's callable

File: src/hometask5/runner.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# runner() – все фукнции вызываются по очереди
+# runner(‘func_name’) – вызывает только функцию func_name.
+# runner(‘func’, ‘func1’...) - вызывает все переданные функции
+
+import the_functions
+
+
+def runner(*functions):
+    if not functions:
+        functions = [el for el in dir(the_functions) if el[:2]
```
Comment (belovmd) [inline src/hometask5/runner.py]: `el.startswith('__')`

File: src/hometask5/the_decorator.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Создайте декоратор, который хранит результаты вызовы
+# функции (за все время вызовов, не только текущий запуск программы)
+
+import the_functions
+
+
+def file_logger_decorator(function):
+    def wrap(*args, **kwargs):
+        function_result = function(*args, **kwargs)
+        with open('call_results.txt', 'a') as f:
```
Comment (belovmd) [inline src/hometask5/the_decorator.py:13]: also we can add date time :)

File: src/hometask5/get_ranges.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Реализовать функцию get_ranges которая получает на вход непустой
+# список неповторяющихся целых чисел, отсортированных по
+# возрастанию, которая этот список “сворачивает”
+# get_ranges([0, 1, 2, 3, 4, 7, 8, 10]) // "0-4,7-8,10"
+# get_ranges([4,7,10]) // "4,7,10"
+# get_ranges([2, 3, 8, 9]) // "2-3,8-9"
+
+
+def get_ranges(lst):
```
Comment (belovmd) [inline src/hometask5/get_ranges.py:9]: great!

File: src/hometask5/IMDB.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
+# В файле хранятся данные с сайта IMDB.
```
Comment (belovmd) [inline src/hometask5/IMDB.py:1]: great

File: src/hometask5/max_divisor.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# Вводится число найти его максимальный делитель,
```
Comment (belovmd) [inline src/hometask5/max_divisor.py:1]: it was task for binary shift ^)

File: src/hometask5/near_power_of_two.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# Написать программу которая находит ближайшую степень двойки
+# к введенному числу. 10(8), 20(16), 1(1)
+
+import math
+
+
+def near_power_of_two(num):
```
Comment (belovmd) [inline src/hometask5/near_power_of_two.py:10]: it was task for binary shift ^)

**Review by belovmd — CHANGES_REQUESTED** [review]
- Please add rating file to gitignore
Great work, please address these issues

File: src/hometask5/runner.py (hunk 1 above)
Comment (ghost) [reply src/hometask5/runner.py]: Fixed

File: src/hometask5/runner.py (hunk 2 above)
Comment (ghost) [reply src/hometask5/runner.py]: Fixed

File: src/hometask5/the_decorator.py (hunk 3 above)
Comment (ghost) [reply src/hometask5/the_decorator.py:13]: The DateTime has been added)

File: src/hometask5/max_divisor.py (hunk 6 above)
Comment (ghost) [reply src/hometask5/max_divisor.py:1]: Made new solution, left and commented the old one

File: src/hometask5/near_power_of_two.py (hunk 7 above)
Comment (ghost) [reply src/hometask5/near_power_of_two.py:10]: Made new solution, left and commented the old one

**Review by belovmd — COMMENTED** [review]
9,5

**Pull request closed without merging**